halo/app/abci.go

```diff
@@ -8,19 +8,30 @@ import (
 	"github.com/omni-network/omni/lib/log"
 
 	abci "github.com/cometbft/cometbft/abci/types"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-type finaliseCallback func(context.Context, *abci.RequestFinalizeBlock, *abci.ResponseFinalizeBlock) error
+type postFinalizeCallback func(sdk.Context) error
```

Comment:
refactored this `PostFinalize` hook to accept cosmos SDK instead of cometBFT req and resp. This aligns the implementation to cosmos standardsn (and allows querying DB).